nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -323,6 +323,36 @@ protected Document resolveReference(DocumentRef docRef) {
         }
     }
 
+    protected Document resolveParentReference(DocumentRef docRef) {
+        if (docRef == null) {
+            throw new IllegalArgumentException("null docRef");
+        }
+        Object ref = docRef.reference();
+        if (ref == null) {
+            throw new IllegalArgumentException("null reference");
+        }
+        int type = docRef.type();
+        switch (type) {
+        case DocumentRef.ID:
+            return getSession().getDocumentByUUID((String) ref).getParent();
+        case DocumentRef.INSTANCE:
+            return getSession().getDocumentByUUID(((DocumentModel) ref).getId()).getParent();
+        case DocumentRef.PATH:
+            Document parentDoc = null;
+            String docPath = (String) ref;
+            if (!"/".equals(docPath)) {
+                String parentPath = docPath.substring(0, docPath.lastIndexOf('/'));
+                if (parentPath.isEmpty()) {
+                    parentPath = "/";
+                }
+                parentDoc = resolveReference(new PathRef(parentPath));
+            }
+            return parentDoc;
+        default:
+            throw new IllegalArgumentException("Invalid type: " + type);
+        }
+    }
```

Comment:
Sounds good, yet, to avoid duplicating most of the the swicth/case code from `#resolveReference`, what do you think of:
```
    protected Document resolveParentReference(DocumentRef docRef) {
        // first let's try to handle a path reference
        if (docRef != null && docRef.type() == DocumentRef.PATH && docRef.reference() != null) {
            String docPath = (String) docRef.reference();
            if ("/".equals(docPath)) {
                return null;
            }
            String parentPath = docPath.substring(0, docPath.lastIndexOf('/'));
            if (parentPath.isEmpty()) {
                parentPath = "/";
            }
            return resolveReference(new PathRef(parentPath));
        }
        return resolveReference(docRef).getParent();
    }
```